### ISSUE TITLE
`TypeAlias` does not work when attribute-accessed from the `typing_extensions` package

### ISSUE THREAD

**Issue by bzoracler** [python/mypy#12179]
**Bug Report**

`TypeAlias` does not trigger errors in `mypy` if accessed from the `typing_extensions` package (rather than being imported directly).

**To Reproduce**

```python
import typing as tp
import typing_extensions as tpx
from typing_extensions import TypeAlias

ASDF: TypeAlias = tp.Literal["A", "S", "D", "F"]
JKL: tpx.TypeAlias = tp.Literal["J", "K", "L"]

Int: tpx.TypeAlias = int
Str: TypeAlias = str
```

**Expected Behavior**

Everything below should throw errors in `mypy`:

```python
G: ASDF = "G"  # error: Incompatible types in assignment (expression has type "Literal['G']", variable has type "Union[Literal['A'], Literal['S'], Literal['D'], Literal['F']]")
H: JKL = "H"  # error: Incompatible types in assignment (expression has type "Literal['H']", variable has type "Union[Literal['J'], Literal['K'], Literal['L']]")
n: Int = "9"  # error: Incompatible types in assignment (expression has type "str", variable has type "int")
s: Str = 9  # error: Incompatible types in assignment (expression has type "int", variable has type "str")
```

**Actual Behavior**

The aliases `JKL` and `Int`, typed using `tpx.TypeAlias`, do not trigger errors with `mypy`:

```python
G: ASDF = "G"  # error: Incompatible types in assignment (expression has type "Literal['G']", variable has type "Union[Literal['A'], Literal['S'], Literal['D'], Literal['F']]")
H: JKL = "H"  # Expected error here; got none
n: Int = "9"  # Expected error here; got none
s: Str = 9  # error: Incompatible types in assignment (expression has type "int", variable has type "str")
```

**Your Environment**

- Mypy version used: `0.931`
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files):
  ```
  [mypy]
  plugins = numpy.typing.mypy_plugin
  ignore_missing_imports = True
  ```
- Python version used: `3.8.12`
- Operating system and version: Elementary OS 6.1 Jólnir (Based on Ubuntu 20.04 LTS)

**Comment by hauntsaninja** [comment]
Yikes, this is pretty bad. I think the issue isn't about typing_extensions vs typing (which tests cover), but rather the import alias. That is, `tp.TypeAlias` fails as well. Looks easy to fix, though!
